gql/dsl.py

```diff
@@ -151,17 +152,37 @@ def serialize_list(serializer, list_values):
     return ListValueNode(values=FrozenList(serializer(v) for v in list_values))
 
 
+class SerializerWrapper:
+    def __init__(self):
+        self._serializer = None
+
+    def set_serializer(self, serializer):
+        self._serializer = serializer
+        return self._serializer
+
+    def __call__(self, *args, **kwargs):
+        return self._serializer(*args, **kwargs)
+
+
+visited_arg_types: Dict[Any, SerializerWrapper] = {}
```

Comment:
I don't like the idea to add a global variable here which is never cleaned. Maybe this could be an attribute of DSLSchema instead ?
This would need a refactor, adding a reference to DSLSchema to the DSLType and DSLField classes, and putting the method get_arg_serializer inside the DSLSchema class ?

Reply:
I've removed the global variable and made the `visited_arg_types` as a required argument with the name `known_serializers` into `get_arg_serializer`. So that we don't need to refactor that many places, but also allowed multiple `get_arg_serializer` invocation within the same context (using one dictionary passed in):
known_serializers = {}
serializer_1 = get_arg_serializer(type_1, known_serializers)
serializer_2 = get_arg_serializer(type_2, known_serializers)
serializer_3 = get_arg_serializer(type_3, known_serializers)

Reply:
This is indeed much better. OK for me